### ISSUE TITLE
Softmax Regression issues

### ISSUE THREAD

**Issue by PrimozGodec** [biolab/orange3#4715]
Softmax Regression has several issues:
1. The shape of the model prediction shape (probability) is wrong when the domain has more values, but not all are present in the training data (iris[:100]). This can be fixed with replacing `fit` function with `fit_storage` - where the domain is passed together with data, or with a general fix in the base class - now a similar fix exists only for skl-learners. I am more for a second option since that way we keep fit function.
2. Softmax regression does not call `__init__` of the base model like other learners do.
